rust/lance-core/src/io/writer/statistics.rs

```diff
@@ -169,8 +685,21 @@ pub fn collect_statistics(arrays: &[&ArrayRef]) -> StatisticsRow {
         DataType::UInt64 => get_statistics::<UInt64Type>(arrays),
         DataType::Float32 => get_statistics::<Float32Type>(arrays),
         DataType::Float64 => get_statistics::<Float64Type>(arrays),
-        DataType::Date32 => get_statistics::<Date32Type>(arrays),
-        DataType::Date64 => get_statistics::<Date64Type>(arrays),
+        DataType::Date32 => get_temporal_statistics(arrays),
+        DataType::Time32(_) => get_temporal_statistics(arrays),
+        DataType::Date64 => get_temporal_statistics(arrays),
+        DataType::Time64(_) => get_temporal_statistics(arrays),
+        DataType::Timestamp(_, _) => get_temporal_statistics(arrays),
+        DataType::Duration(_) => get_temporal_statistics(arrays),
```

Comment:
BTW this can be simplified to:

```suggestion
        DataType::Date32 | DataType::Time32(_) | DataType::Date64 | DataType::Time64(_)
        | DataType::Timestamp(_, _) | DataType::Duration(_) => get_temporal_statistics(arrays),
```